python_modules/dagster/dagster/core/definitions/partition.py

```diff
@@ -161,6 +161,11 @@ class PartitionsDefinition(ABC, Generic[T]):
     def get_partitions(self, current_time: Optional[datetime] = None) -> List[Partition[T]]:
         ...
 
+    def __str__(self) -> str:
+        partitioned_keys = self.get_partition_keys()
+        joined_keys = ', '.join([f"'{key}'" for key in partitioned_keys])
+        return f"Partitioned by {'key' if len(partitioned_keys) == 1 else 'keys'} {joined_keys}."
```

Comment:
Imagining this showing up here:
![image](https://user-images.githubusercontent.com/654855/149039403-9c644d8f-7592-4c69-860b-2d814e76d914.png)

It might look a little weird to have "Partitioned by" underneath "Partitions".

Reply:
I modified this to only contain the partition strings: e.g. "'foo', 'bar', 'baz'"